analyzers/tests/SonarAnalyzer.Test/TestCases/RedundancyInConstructorDestructorDeclaration.CSharp12.cs

```diff
@@ -3,9 +3,30 @@ namespace Repro_8092
 {
     namespace PrimaryParameterlessConstructor
     {
-        class AClassWithBody() { }                // Noncompliant {{Remove this redundant primary constructor.}}
+        class AClassWithBody() { }                                              // Noncompliant {{Remove this redundant primary constructor.}}
 //                          ^^
-        class AClassWithoutBody();                // Noncompliant
+        class AClassWithoutBody();                                              // Noncompliant
+
+        public abstract class BaseClass(string value);
+
+        public class SomeClass() : BaseClass("SomeValue");                      // Compliant, Foo is calling Base constructor with primary constructor syntax, it's not redundant
+
+        public class DefaultStyle : BaseClass
+        {
+            public DefaultStyle() : base("SomeValue")                           // Compliant, "default" way of calling Base constructor
+            { }
+        }
```

Comment:
Add a noncompliant test case that uses a base class constructor without parameters.